gulpfile.js

```diff
@@ -136,60 +138,47 @@ gulp.task('sass', () => {
-// It will start watching for changes in every .js file.
-// If there's a change, the task 'scripts' defined above will fire.
-gulp.task('default', sequence(
-    ['clean-js', 'clean-css'],
-    ['copy:index.html', 'copy:fonts', 'sass', 'scripts', 'backend'],
-    'index',
-    'connect',
-    ['watch', 'watch:backend']
-));
-
-// Private Functions
-// ----------------------------------------------------------------------------
-
-function bundleApp(isProduction) {
-    scriptsCount++;
+gulp.task('scripts:vendor', (done) => {
+    browserify({
+        require: dependencies,
+        debug: ENVIRONMENT.isDev,
+        extensions: CONFIG.extensions
+    })
+        .bundle()
+        .on('error', gutil.log)
+        .pipe(source('vendors.js'))
+        .pipe(gulp.dest(CONFIG.tempFolder));
+    done();
+});
 
-    // Browserify will bundle all our js files together in to one and will let
-    // us use modules in the front end.
+gulp.task('scripts:app', (done) => {
     const appBundler = browserify({
         entries: CONFIG.appEntryPoint,
-        debug: true,
+        debug: ENVIRONMENT.isDev,
         extensions: CONFIG.extensions
     });
 
-    // If it's not for production, a separate vendors.js file will be created
-    // the first time gulp is run so that we don't have to rebundle things like
-    // react everytime there's a change in the js file
-    if (!isProduction && scriptsCount === 1) {
-        // create vendors.js for dev environment.
-        browserify({
-            require: dependencies,
-            debug: true,
-            extensions: CONFIG.extensions
-        })
-            .bundle()
-            .on('error', gutil.log)
-            .pipe(source('vendors.js'))
-            .pipe(gulp.dest(CONFIG.tempFolder));
-    }
-    if (!isProduction) {
-        // make the dependencies external so they dont get bundled by the
-        // app bundler. Dependencies are already bundled in vendor.js for
-        // development environments.
-        dependencies.forEach(function (dep) {
-            appBundler.external(dep);
-        });
-    }
+    dependencies.forEach(dep => {
+        appBundler.external(dep);
+    });
 
     appBundler
         // transform ES6 and JSX to ES5 with babelify
-        .transform('babelify', {presets: ['es2015', 'react']})
+        .transform('babelify', { presets: ['es2015', 'react'] })
         .bundle()
         .on('error', gutil.log)
         .pipe(source('bundle.js'))
         .pipe(gulp.dest(CONFIG.tempFolder));
-}
+    done();
+});
+
+// When running 'gulp' on the terminal this task will fire.
+// It will start watching for changes in every .js file.
+// If there's a change, the task 'scripts' defined above will fire.
+gulp.task('default', sequence(
+    ['clean-js', 'clean-css'],
+    ['copy:index.html', 'copy:fonts', 'sass', 'scripts:vendor', 'scripts:app', 'backend'],
+    'index',
+    'connect',
+    ['watch', 'watch:backend']
+));
```

Comment:
For the sake of consistency, please rename "clean-js" to "clean:js" and "clean-css" to "clean:css".
Also, we have a task named _index_, so why not name the other "copy:index"?
#niceToHave